AppWidget/app/src/main/res/values-v31/styles.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2021 The Android Open Source Project
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<resources>
+    <!--
+    Having clipToOutline to true is needed for the background (that has id with
+    android:id/background so that the system knows the widget has been updated
+    and doesn't need to be clipped by the launcher.
+    -->
+    <style name="Widget.AppWidget.AppWidget.Container" parent="android:Widget">
+        <item name="android:padding">?attr/appWidgetPadding</item>
+        <item name="android:background">@drawable/app_widget_background</item>
+        <item name="android:clipToOutline">true</item>
```

Comment:
Potentially silly question: Can we just add this to values-21/styles.xml? It looks like clipToOutline was [added in API 21](https://developer.android.com/reference/android/view/View#getClipToOutline())

Reply:
It's not necessary as AAPT2 will handle this automatically.

Reply:
It's not a silly question :) I originally had the same question. Although clipToOutline was introduced in API 21, the XML attribute was introduced in API 31.

Reply:
Thanks for explaining!